patches/sitecustomize.py

```diff
@@ -1,23 +1,29 @@
 import os
-
+from kaggle_gcp import get_integrations
 kaggle_proxy_token = os.getenv("KAGGLE_DATA_PROXY_TOKEN")
+kernel_integrations_var = os.getenv("KAGGLE_KERNEL_INTEGRATIONS")
+
 bq_user_jwt = os.getenv("KAGGLE_BQ_USER_JWT")
 if kaggle_proxy_token or bq_user_jwt:
     from google.auth import credentials
     from google.cloud import bigquery
     from google.cloud.bigquery._http import Connection
     # TODO: Update this to the correct kaggle.gcp path once we no longer inject modules
     # from the worker.
-    from kaggle_gcp import PublicBigqueryClient
+    from kaggle_gcp import PublicBigqueryClient, KaggleKernelCredentials
 
     def monkeypatch_bq(bq_client, *args, **kwargs):
         data_proxy_project = os.getenv("KAGGLE_DATA_PROXY_PROJECT")
         specified_project = kwargs.get('project')
         specified_credentials = kwargs.get('credentials')
-        if specified_project is None and specified_credentials is None:
+        kernel_integrations = get_integrations()
+        if specified_project is None and specified_credentials is None and not kernel_integrations.has_bigquery():
             print("Using Kaggle's public dataset BigQuery integration.")
```

Comment:
I was sort of in favour of adding this to make the user more explicitly aware of what is happening, and don't think the console log line is that intrusive, but curious on opinion from others.

Reply:
This one shows up for me in the Notebook outptut, i.e. imagine the user has some code like the following:
```
datasets = list(client.list_datasets())
project = client.project
if datasets:
    print("Datasets in project {}:".format(project))
```
These the debug msg ("Using Kaggle's public dataset..") shows up for me in the output, where I'd expect the output to just contain my own print statements. 

Do you see what I mean?

Reply:
Right, that's what I meant by console log line, it shows in the output. I can see both sides of this: on the one hand libraries don't typically print out statements to the console, but on the other hand it may provide users with indication of what is happening (and may help resolve more issues than it may cause). How do Kernels folks feel about this? Is there a better approach?

Reply:
Given that it only prints if a user uses BigQuery, I am fine with it. It will help user understand what is going on and reduce their support need.

Reply:
I know this a minor point that shouldn't affect the PR review, but I'm just pointing out things that stood out for me :). Again, please _feel free to ignore this minor point._ 

Printing in a library is generally considered not a good practice to my knowledge. Instead, for debugging purpose, we could have [just used logging](https://www.google.com/search?q=python+logging+vs+print&oq=python+logging+vs+print)? 

If I have function in my Notebook that uses bigquery library, and I call it from my Notebook, I'll see these print statements (possibly multiple times?) - to me, this is unexpected.